### DISCUSSION TITLE
Truncate array

### DISCUSSION THREAD

**Discussion by nikitamasych** [Consensys/gnark#809] [Q&A] [answered]
Hi! I have `[32]frontend.Variable` input in circuit, as well as the end (pivot) index. I want to receive slice of elements up to this pivot. Functionality analogous to `selector.Slice`, but instead of zeroing elements after the pivot, I need to cut the array length. Is it technically achievable?

**Comment by ivokub** [comment, marked as answer]
Do you mean that the length of the returned slice depends on pivot? Unfortunately, it is not possible - gnark allocates the arrays (i.e. the variables contained within) at compile time, when we do not yet know the value for pivot. Only during solving time (within prover) are the variables actually solved.
